### ISSUE TITLE
Using correto ACCP with  Bouncy castle for TLS in Fips mode issues

### ISSUE THREAD

**Issue by chavdarch** [corretto/amazon-corretto-crypto-provider#291]
Hi,

My company is trying to use ACCP for JCA/JCE and Bouncy Castle for TLS in Fips mode, but we are facing issues with the DH key exchange see https://github.com/bcgit/bc-java/issues/1369.
It looks like BC don't like the FIPs secure random provided by ACCP and TLS connections are failing.

Do you have any example/recommendation on using ACCP in FIPs mode along with FIPs-enabled TLS(provided by Bouncy Castle)

**Comment by geedo0** [comment]
Hi there, thanks for bringing this up. We made a [change very recently](https://github.com/corretto/amazon-corretto-crypto-provider/pull/286) which made the registration of an ACCP-backed SecureRandom configurable. For the FIPS build, we actually chose to default this to false which is exactly what you need in this case (i.e. BC-FIPS' TLS won't hit the ACCP provided RNG since it won't be there). Can you try pulling in the latest changes and letting us know if that works for you?

**Comment by WillChilds-Klein** [comment]
just as an observation, I noticed something odd in the linked ticket's stacktrace:

```
Caused by: org.bouncycastle.crypto.fips.FipsUnapprovedOperationError: Attempt to create key pair with unapproved RNG: DH
	at org.bouncycastle.fips.core/org.bouncycastle.crypto.fips.Utils.validateRandom(Unknown Source)
	at org.bouncycastle.fips.core/org.bouncycastle.crypto.fips.Utils.validateKeyPairGenRandom(Unknown Source)
	at org.bouncycastle.fips.core/org.bouncycastle.crypto.fips.FipsDH$KeyPairGenerator.<init>(Unknown Source)
```

`DH` isn't the name of any SecureRandom algorithm provided by java nor by ACCP. according to the [JCA docs](https://docs.oracle.com/javase/8/docs/technotes/guides/security/StandardNames.html#KeyPairGenerator) `DH` is, however, the algorithm name returned by [Key.getAlgorithm()](https://docs.oracle.com/javase/8/docs/api/java/security/Key.html#getAlgorithm--). this makes me wonder if some wires are getting crossed somewhere in BC JCA or JSSE provider(s) such that a Key instance's algorithm is being used to request a particular flavor of SecureRandom algorithm (e.g. `SHA1PRNG`, `NativePRNG`, `DEFAULT`, ACCP's `LibCryptoRng`, etc.).

**Comment by chavdarch** [comment]
Thank you so much!
We will try your recommendations and post the results here.

**Comment by chavdarch** [comment]
We can confirm that [your change](https://github.com/corretto/amazon-corretto-crypto-provider/pull/286) has indeed fixed the problem for us. Thank you so much!

When do you think you will be able to make a release with that change?

**Comment by WillChilds-Klein** [comment]
Glad to hear that worked for  you @chavdarch . We're putting together a release to Maven Central for next week. Please keep an eye out [here](https://central.sonatype.com/artifact/software.amazon.cryptools/AmazonCorrettoCryptoProvider/2.0.0/versions) for 2.0.1.

**Comment by WillChilds-Klein** [comment]
These changes were [released in 2.1](https://github.com/corretto/amazon-corretto-crypto-provider/releases/tag/2.1.0). Please re-open this issue if needed.
